app/controllers/integrated/who_is_married_controller.rb

```diff
@@ -0,0 +1,29 @@
+module Integrated
+  class WhoIsMarriedController < FormsController
+    def self.skip?(application)
+      return true if application.single_member_household?
+      return true unless application.navigator.anyone_married?
+    end
+
+    def edit
+      @form = form_class.new(members: current_application.members)
+    end
+
+    def update
```

Comment:
Here's the relevant logic that I think we could tackle refactoring now (I can tackle between meetings tomorrow) or in the future. It's really similar to the update method in `DescribeTaxRelationshipsController`